addons/l10n_ch/models/account_invoice.py

```diff
@@ -233,19 +233,21 @@ def _compute_l10n_ch_isr_optical_line(self):
     @api.depends(
         'type', 'name', 'currency_id.name',
         'invoice_partner_bank_id.l10n_ch_isr_subscription_eur',
-        'invoice_partner_bank_id.l10n_ch_isr_subscription_chf')
+        'invoice_partner_bank_id.l10n_ch_isr_subscription_chf',
+        'company_id.country_id.code')
     def _compute_l10n_ch_isr_valid(self):
         """Returns True if all the data required to generate the ISR are present"""
         for record in self:
             record.l10n_ch_isr_valid = record.type == 'out_invoice' and\
                 record.name and \
                 record.l10n_ch_isr_subscription and \
-                record.l10n_ch_currency_name in ['EUR', 'CHF']
+                record.l10n_ch_currency_name in ['EUR', 'CHF'] and \
+                record.company_id.country_id.code in ["CH", "LI"]
 
     @api.depends('type', 'invoice_partner_bank_id', 'invoice_payment_ref')
     def _compute_l10n_ch_isr_needs_fixing(self):
         for inv in self:
-            if inv.type == 'in_invoice' and inv.company_id.country_id.code == "CH":
+            if inv.type == 'in_invoice' and inv.company_id.country_id.code in ["CH", "LI"]:
```

Comment:
@yvaucher Here it makes sense, but I don't see immediately the added value in v13 for the other cases where you add it. Does LI also need cash rounding, ...  Are they happy enough using l10n_ch?

Reply:
I have no idea, I just know they use QR-bill. And if they use QR-bill they might need the reference type ISR.

Here the aim is mostly to not have the QR-bill attached to France, Germany or Belgium (for example) companies when installed on the same database as a company in Switzerland. It messes up the mail sent.

So I reduced
* _all countries of the world_
to
* _the 2 countries that uses QR-bill_

Then updated other country code check to be consistent. (but the one in deprecated function)

I don't even know if there are user in Liechtenstein. (I hope there are :) )

From [Swiss Implementation Guidelines forthe QR-bill](https://www.paymentstandards.ch/dam/downloads/ig-qr-bill-en.pdf)

> It should also be noted that the QR-bill is based on the ISO20022 standard and was designed primarily for invoicing in Switzerland and Liechtenstein. [...]

IMO, it's another reason why the QR-bill should not be part of `l10n_ch` which should only include the CoA.